### PR TITLE
luminous: tools: ceph-objectstore-tool: Dump hashinfo

### PR DESCRIPTION
**Pull request by ashishkumsingh**
https://tracker.ceph.com/issues/37690

### CODE REVIEW

File: src/tools/ceph_objectstore_tool.cc [hunk 1]
```diff
@@ -2399,6 +2400,22 @@ int print_obj_info(ObjectStore *store, coll_t coll, ghobject_t &ghobj, Formatter
       formatter->close_section();
     }
   }
+  bufferlist hattr;
+  gr = store->getattr(ch, ghobj, ECUtil::get_hinfo_key(), hattr);
```
Comment (tchaikov) [inline src/tools/ceph_objectstore_tool.cc]: i think collection handle is a relatively new thing -- it was introduced in mimic. probably you could take a look 38d24fefd6fa80c9f85afc9e0e4570b6fe387a7e for some inspirations?

**Comment by tchaikov** [comment]
@ashishkumsingh the backport does not compile. please see my comment.

File: src/tools/ceph_objectstore_tool.cc (hunk 1 above)
Comment (ashishkumsingh) [reply src/tools/ceph_objectstore_tool.cc]: @tchaikov yes, thank you for the pointers. I have replaced the 'ch' to 'coll' as per previous calls to 'store->getattr' and also changed the 'cbegin()' to 'begin()'.

**Comment by dzafman** [comment]
@ashishkumsingh Did you run the ceph-objectstore-tool comand in a vstart.sh with an erasure coded pool with an object or two?

**Comment by dzafman** [comment]
I have a rudimentary test for this in ceph/ceph#26158 in master.  We could backport that pull request in combination with this one.

**Comment by dzafman** [comment]
Testing this requires using test in ceph/ceph#26284

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-02-11-1725-luminous](https://trello.com/c/82sp4VSF/535-wip-yuri4-testing-2019-02-11-1725-luminous)

**Pull request merged**